server/src/main/java/org/opensearch/action/admin/indices/create/TransportCreateIndexAction.java

```diff
@@ -91,7 +91,13 @@ protected CreateIndexResponse read(StreamInput in) throws IOException {
 
     @Override
     protected ClusterBlockException checkBlock(CreateIndexRequest request, ClusterState state) {
```

Comment:
For my understanding, can you please explain what does `checkBlock` function do?

Reply:
It is called from the base class [TransportClusterManagerNodeAction](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/support/clustermanager/TransportClusterManagerNodeAction.java#L192). Its purpose is to check if there is any block which prevents index creation.